### PR TITLE
delete_deprecated_content_from_conf

### PR DESCRIPTION
**Pull request by sapirshuker**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3573

## Description
When deprecating a pack, we need to also remove its test playbook from test.conf file. Adding this step to the -d flag will help in not forgetting to do this important step when deprecating an integration.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 * Improved memory efficiency in **update-content-graph** and **create-content-graph** commands.
 * Removed support for the `cve_id` name for the default-argument for **cve** reputation commands in **validate**. Now, only `cve` may be used for such commands.
 * Updated *Pydantic* to v2.
+* Added support for **format** with the `-d` flag to delete TestPlaybooks from `conf.json`.
```
Comment (Shellyber) [inline CHANGELOG.md:6]:
```suggestion
* Added to the **format** command the ability to remove test playbooks from `conf.json` file when deprecating content items.
```

File: demisto_sdk/commands/common/tools.py [hunk 2]
```diff
@@ -2250,6 +2251,101 @@ def is_test_config_match(
     return False
 
 
+def is_content_item_dependent_in_conf(test_config, file_type):
+    """Check if a line from conf have multiple integration/scripts dependent on the TPB.
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tools.py:2255]: Let's add example here for the 2 cases: one with 2 dependent and second case with only one (regular)

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -2250,6 +2251,101 @@ def is_test_config_match(
     return False
 
 
+def is_content_item_dependent_in_conf(test_config, file_type):
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tools.py:2254]: Add return type hints in all functions

File: demisto_sdk/commands/common/tools.py [hunk 4]
```diff
@@ -2250,6 +2251,101 @@ def is_test_config_match(
...
+        content_item_id (str): The  content item id.
+        file_type (str): The file type, can be integrations, scripts or playbook.
+        test_playbooks (list): A list of related test playbooks.
+        no_test_playbooks_explicitly (bool): True if there are related TPB, False otherwise.
+
+    Returns:
+        bool: The return value. True for dependence, False otherwise.
+    """
+    test_registered_in_conf_json = []
+    # if the file time that we are deprected is a integration there are TBP related to the yml
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tools.py:2300]:
```suggestion
    # If the file type we are deprecating is a integration - there are TBP related to the yml
```

File: demisto_sdk/commands/common/tests/tools_test.py [hunk 5]
```diff
@@ -2851,3 +2853,194 @@ def test_sha1_dir():
...
+    """
+    CONF_JSON_ORIGINAL_CONTENT = {
+        "tests": [
+            {"integrations": ["PagerDuty v1"], "playbookID": "PagerDutyV1 Test"},
+            {
+                "integrations": ["PagerDuty v1", "PagerDuty v2"],
+                "playbookID": "PagerDuty Test",
+            },
+            {"integrations": "PagerDuty v2", "playbookID": "PagerDutyV2 Test"},
+            {"integrations": "PagerDuty v2", "playbookID": "PagerDutyV2 Test"},
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tests/tools_test.py:2970]: Duplicate?

File: demisto_sdk/commands/format/update_generic_yml.py [hunk 6]
```diff
@@ -259,6 +269,34 @@ def update_tests(self) -> None:
                 else:
                     logger.debug(f'Not formatting {self.source_file} with "No tests"')
 
+    def remove_from_conf_json(self, file_type, content_item_id) -> None:
```
Comment (Shellyber) [inline demisto_sdk/commands/format/update_generic_yml.py:272]: Let's add test to the format command as well. 
when using the -d flag, that it does what it suppose to do.

File: demisto_sdk/commands/format/update_generic_yml.py [hunk 7]
```diff
@@ -259,6 +269,34 @@ def update_tests(self) -> None:
...
+        """
+        Updates conf.json remove the file's test playbooks.
+        Args:
+            file_type: The typr of the file, can be integration, playbook or testplaybook.
+            content_item_id: The content item id.
+        """
+        related_test_playbook = self.data.get("tests", [])
+        no_test_playbooks_explicitly = any(
+            test for test in related_test_playbook if "no test" in test.lower()
```
Comment (Shellyber) [inline demisto_sdk/commands/format/update_generic_yml.py:281]: Also need to check for `no tests`

**Review by Shellyber — CHANGES_REQUESTED** [review]
Great work.
- Please also schedule a demo with me to go over everything 
- Add Return type hints where missing
- Add test also to the format command

File: demisto_sdk/commands/common/tests/tools_test.py (hunk 5 above)
Comment (sapirshuker) [reply demisto_sdk/commands/common/tests/tools_test.py:2970]: Yes, it is on purpose, in order to check adage cases.

File: CHANGELOG.md (hunk 1 above)
Comment (dorschw) [inline CHANGELOG.md:6]:
```suggestion
* Calling **format** with the `-d` flag now removes test playbooks testing the deprecated content from conf.json.
```

File: demisto_sdk/commands/common/tools.py [hunk 8]
```diff
@@ -2244,9 +2257,129 @@ def is_test_config_match(
...
+        file_type (str): The file type, can be integrations, scripts or playbook.
+        test_playbooks (list): A list of related test playbooks.
+        no_test_playbooks_explicitly (bool): True if there are related TPB, False otherwise.
+
+    Returns:
+        bool: The return value. True for dependence, False otherwise.
+    """
+    test_registered_in_conf_json = []
+    # If the file type we are deprecating is a integration - there are TBP related to the yml
+    if file_type == "integration":
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tools.py:2326]: Too much duplicate code here. Let's do something like this:
```
if file_type in ["integration", "playbook"]:
    if file_type == "integration":
        content_item_id = "integration_id"
    else:  # file_type == "playbook"
        content_item_id = "test_playbook_id"

    test_registered_in_conf_json.extend(
        [
            test_config
            for test_config in conf_json_tests
            if is_test_config_match(test_config, **{keyword: content_item_id})
        ]
    )

```
For all use cases

**Review by Shellyber — COMMENTED** [review]
Nice work

File: demisto_sdk/commands/common/tools.py [hunk 9]
```diff
@@ -2223,9 +2236,113 @@ def is_test_config_match(
...
+
+    Returns:
+        bool: The return value. True for dependence, False otherwise.
+    """
+    integrations_list = test_config.get("integrations", [])
+    integrations_list = (
+        integrations_list
+        if isinstance(integrations_list, list)
+        else [integrations_list]
+    )
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tools.py:2261]:
```suggestion
integrations_list = integrations_list if isinstance(integrations_list, list) else [integrations_list]
```

Comment (Shellyber) [reply demisto_sdk/commands/common/tools.py:2261]: No need for all the extra lines here, ignore the black if necessary
same for scripts

File: demisto_sdk/commands/common/tools.py [hunk 10]
```diff
@@ -2223,9 +2236,113 @@ def is_test_config_match(
...
 
+    if script_id:
+        return scripts_match
+
     return False
 
 
+def is_content_item_dependent_in_conf(test_config, file_type) -> bool:
+    """Check if a line from conf have multiple integration/scripts dependent on the TPB.
+        - if the TPB checks only one integration/script it is independent.
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tools.py:2247]: And give example for each case

File: demisto_sdk/commands/common/tools.py [hunk 11]
```diff
@@ -2223,9 +2236,113 @@ def is_test_config_match(
...
+
+    Returns:
+        bool: The return value. True for dependence, False otherwise.
+    """
+    keyword = ""
+    test_registered_in_conf_json = []
+    # If the file type we are deprecating is a integration - there are TBP related to the yml
+    if file_type == "integration":
+        keyword = "integration_id"
+    # the type of the file is "playbook"
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tools.py:2300]:
```suggestion
```

File: demisto_sdk/commands/common/tools.py [hunk 12]
```diff
@@ -2223,9 +2236,113 @@ def is_test_config_match(
...
+    elif file_type == "playbook":
+        keyword = "test_playbook_id"
+
+    # the type of the file is "script"
```
Comment (Shellyber) [inline demisto_sdk/commands/common/tools.py:2304]:
```suggestion
```

**Review by Shellyber — APPROVED** [review]
Nice work

**Pull request merged**